### PR TITLE
add more e2e tests

### PR DESCRIPTION
**Pull request by BuckyMaler**
`NODE_ENV` needs to be set to `production` to test a production build because `jest` sets it to `test` by default. I added [`cross-env`](https://www.npmjs.com/package/cross-env) so that setting `NODE_ENV` works across all platforms.

### CODE REVIEW

**Comment by BuckyMaler** [comment]
@ZachJW34 please rerun the CI build. I don't have permission to.

**Comment by BuckyMaler** [comment]
I increased the CircleCI timeout because with these changes running our e2e tests exceeds CircleCI's default 10 minute no-output timeout. This issue is described in more detail [here](https://support.circleci.com/hc/en-us/articles/360007188574-Build-has-hit-timeout-limit).

File: apps/vue-e2e/tests/vue.test.ts [hunk 1]
```diff
@@ -3,43 +3,153 @@ import {
...
+        e2e: false,
+        build: true,
+        buildProd: true,
+      });
+
+      done();
+    }, 300000);
+  });
+
+  it('should generate app with routing and add vuex', async (done) => {
```
Comment (ZachJW34) [inline apps/vue-e2e/tests/vue.test.ts:127]: What's the value of this test? Cover all bases?

Comment (BuckyMaler) [reply apps/vue-e2e/tests/vue.test.ts:127]: This test doesn't follow our pattern of testing one option at a time, but I feel the majority of apps will begin with routing and Vuex and there's overlap (e.g. `main.ts`) when they're used together so I wanted to help ensure that combination doesn't break.

**Pull request merged**